### ISSUE TITLE
Autocomplete dev help in package development `?fn`

### ISSUE THREAD

**Issue by olivroy** [rstudio/rstudio#13526]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2023.06.1
    OS Version      :  Windows
    R Version       : 4.2.3

### Steps to reproduce the problem

1. Open a package in RStudio
2. Run `library(devtools)`
3. Run `devtools::load_all()`
4. Type `?`
5. Type Tab (or wait for autocompletion to start)
6. The suggestions are unrelated to the current package

### Describe the problem in detail

In package dev, the `?` takes very long to autocomplete. Also, it shows the loaded packages functions instead of the functions available via `load_all()`. It doesn't try to autocomplete the important functions. It autocompletes the loaded packages instead. (testthat, devtools,, etc.)

 Because running `load_all()` overrides the `?` method.

Normally, utils provides `?`, but in pkg dev, pkgload provides the `?` function. related to #13460

![image](https://github.com/rstudio/rstudio/assets/52606734/d78b0c5e-4b90-462a-9dff-d3e821eeca61)

### Describe the behavior you expected

After `load_all()`, I would expect autocompletion to priorize objects loaded with load_all when autocompleting `?`.

Autcompletion works as expected (in the correct package dev environment) when using the function, so I would expect autocomplete to work when trying to visualize the dev help.

I would expect the autocomplete for help to behave similarly to autocomplete for the function itself. It would be very helpful to visualize dev-help.
![image](https://github.com/rstudio/rstudio/assets/52606734/75c9c407-118d-4517-8ab5-675f08dc3b03)

In short, I would expect `?` autocompletes and regular autocomplete to behave the same.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@olivroy Thank you for raising this! I'll mark it for review.

**Comment by olivroy** [comment]
The help shows up in dev! That's good. However, if I am not mistaken, the autocomplete seems to be with the installed version of the package rather than the dev version of the function.

There are 2 questions however,

if you change the documentation and re-document, the popup doesn't change. 
If you change an argument to your function, the popup doesn't work anymore.

But tbh, the help popup was above my expectations, so I wouldn't mind closing this as complete.


But it is still great in 2023.12.0+76!

the F1 shortcut doesn't work, but I guess it's fine. At least, now you don't have to type the whole function :) 
![image](https://github.com/rstudio/rstudio/assets/52606734/fb9c2fa1-a9f7-4ebc-bff9-718237d9e013)

**Comment by ronblum** [comment]
@olivroy Thanks! Can you please open a new issue for this, since the original issue is resolved?

**Comment by olivroy** [comment]
My initial assessment may not have been correct. It is just that pkgload caches the state at a certain time, but this cache is not updated frequently, which means that if you add a new function or modifies its documentation, it is not taken into account for autocompletion, or in the popup.

The help preview in the help pane appears correct however.
